pylasu/model/model.py

```diff
@@ -70,12 +73,67 @@ def is_internal_property_or_method(value):
     return isinstance(value, internal_property) or isinstance(value, InternalField) or isinstance(value, Callable)
 
 
-class Node(Origin, Destination):
+class Multiplicity(enum.Enum):
+    OPTIONAL = 0
+    SINGULAR = 1
+    MANY = 2
+
+
+@dataclass
+class PropertyDescriptor:
+    name: str
+    provides_nodes: bool
+    multiplicity: Multiplicity = Multiplicity.SINGULAR
+
+    def multiple(self):
+        return self.multiplicity == Multiplicity.MANY
+
+
+def provides_nodes(decl_type):
+    return isinstance(decl_type, type) and issubclass(decl_type, Node)
+
+
+class Concept(ABCMeta):
```

Comment:
With LionWeb we may want to introduce Concept with a close but perhaps different meaning

Reply:
Do we have a corresponding `Concept` in Kolasu?

Reply:
For now Concept does not exist in Kolasu

Reply:
No, at least not explicitly. In Kolasu the concept is implicitly defined by extension methods using reflection. We could have done the same in Pylasu, but the IDE doesn't give any support for Python extension methods, so it's better to be more explicit if possible.